### PR TITLE
Nitish

### PR DESCRIPTION
**Pull request by Nitish0007**
redux in login and signup is done

### CODE REVIEW

File: src/pages/Merchant/LoginPage/LoginPage.js [hunk 1]
```diff
@@ -10,12 +13,16 @@ import { login } from "api/user/login";
...
-import toast from "react-hot-toast";
 
 function LoginPage() {
   const emailRef = useRef();
   const passRef = useRef();
   const [errors, setErrors] = useState({});
+  const navigate = useNavigate();
+
+  // const isMrchantLoggedIn = useSelector((state) => state.merchantAuth);
+  // const merchant = useSelector((state) => state.currMerchant);
```
Comment (aashu0148) [inline src/pages/Merchant/LoginPage/LoginPage.js]: why this comment?

File: src/pages/Merchant/LoginPage/LoginPage.js [hunk 2]
```diff
@@ -41,6 +48,18 @@ function LoginPage() {
...
 
+  const handleNavigate = () => {
+    navigate("/merchant/register");
+  };
+
+  const handleMerchantAuth = (merchantObj) => {
+    dispatch({
+      type: "IS_MERCHANT_LOGGEDIN",
+      merchantAuth: true,
+      currMerchant: merchantObj,
```
Comment (aashu0148) [inline src/pages/Merchant/LoginPage/LoginPage.js]: dont store complete details of user, instead store name,email etc

File: src/pages/Merchant/LoginPage/LoginPage.js [hunk 3]
```diff
@@ -52,7 +71,14 @@ function LoginPage() {
     };
     login(loginDetails).then(async (res) => {
       if (!res) return;
-      toast.success("Logged in successfully");
+      else {
+        if (res?.status) {
+          console.log(res);
```
Comment (aashu0148) [inline src/pages/Merchant/LoginPage/LoginPage.js:76]: why this console?

File: src/pages/Merchant/Register/Register.js [hunk 4]
```diff
@@ -25,12 +27,22 @@ function Register() {
...
 
   const changeURl = () => {
     navigate("/merchant/login");
   };
 
+  const handleMerchantAuth = (merchantObj) => {
+    dispatch({
+      type: "IS_MERCHANT_LOGGEDIN",
+      merchantAuth: true,
+      currMerchant: merchantObj,
```
Comment (aashu0148) [inline src/pages/Merchant/Register/Register.js]: dont store complete obj as above said

File: src/store/merchant/merchant.js [hunk 5]
```diff
@@ -1,10 +1,20 @@
 const initialState = {
   name: "",
+  merchantAuth: false,
+  currMerchant: {},
```
Comment (aashu0148) [inline src/store/merchant/merchant.js]: remove this as said above

File: src/store/merchant/merchant.js [hunk 6]
```diff
@@ -1,10 +1,20 @@
...
 };
 
 const merchantReducer = (state = initialState, action) => {
   switch (action.type) {
-    default : return state
+    case "IS_MERCHANT_LOGGED": {
+      return {
+        ...state,
+        merchantAuth: action.merchantAuth,
+        currMerchant: action.currMerchant,
```
Comment (aashu0148) [inline src/store/merchant/merchant.js]: same need to change this

File: src/pages/Merchant/LoginPage/LoginPage.js (hunk 1 above)
Comment (Nitish0007) [reply src/pages/Merchant/LoginPage/LoginPage.js]: removed

**Comment by Nitish0007** [comment]
navbar design is completed and conflict relsolved . console was not removed becoz i m working on it so i will remove them after completion so dont consider them for now

File: src/components/Navbar/Navbar.js [hunk 7]
```diff
@@ -1,16 +1,54 @@
...
+          </span>
+        </div>
+        <Searchbar />
+        <ShoppingCart />
+        <div className={styles.authSection}>
+          {!props.auth ? (
+            <>
+              <Link to="/merchant/register">Signup</Link>
+              <span>/</span>
+              <Link to="/merchant/login">Login</Link>
```
Comment (aashu0148) [inline src/components/Navbar/Navbar.js:33]: why these link points to only merchant signup login ?

dont change it now but just we need to discuss on it

File: src/components/Navbar/Navbar.js [hunk 8]
```diff
@@ -1,16 +1,54 @@
...
+            </>
+          ) : (
+            <>
+              <div className={styles.user}>
+                <img src="https://cdn-icons-png.flaticon.com/128/145/145843.png" alt="" />
```
Comment (aashu0148) [inline src/components/Navbar/Navbar.js:38]: why using  links directly ??

**Pull request merged**